tools/wpt/browser.py

```diff
@@ -1631,117 +1659,98 @@ def find_binary(self, venv_path=None, channel=None):
+
+        return edgedriver_path
+
+    def install_webdriver_by_version(self, version, dest, channel):
+        self._remove_existing_edgedriver_binary(dest)
+
         if self.platform == "linux":
             bits = "linux64"
-            edgedriver_path = os.path.join(dest, self.edgedriver_name)
         elif self.platform == "macos":
             bits = "mac64"
-            edgedriver_path = os.path.join(dest, self.edgedriver_name)
-        else:
+        elif self.platform == "win":
             bits = "win64" if uname[4] == "x86_64" else "win32"
-            edgedriver_path = os.path.join(dest, f"{self.edgedriver_name}.exe")
-        url = f"https://msedgedriver.azureedge.net/{version}/edgedriver_{bits}.zip"
-
-        # cleanup existing Edge driver files to avoid access_denied errors when unzipping
-        if os.path.isfile(edgedriver_path):
-            # remove read-only attribute
-            os.chmod(edgedriver_path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO)  # 0777
-            print(f"Delete {edgedriver_path} file")
-            os.remove(edgedriver_path)
-        driver_notes_path = os.path.join(dest, "Driver_notes")
-        if os.path.isdir(driver_notes_path):
-            print(f"Delete {driver_notes_path} folder")
-            rmtree(driver_notes_path)
 
+        url = f"https://msedgedriver.azureedge.net/{version}/edgedriver_{bits}.zip"
         self.logger.info(f"Downloading MSEdgeDriver from {url}")
         unzip(get(url).raw, dest)
-        if os.path.isfile(edgedriver_path):
-            self.logger.info(f"Successfully downloaded MSEdgeDriver to {edgedriver_path}")
-        return which(self.edgedriver_name, path=dest)
+        edgedriver_path = which("msedgedriver", path=dest)
+        assert edgedriver_path is not None
+        return edgedriver_path
 
-    def install_webdriver(self, dest=None, channel=None, browser_binary=None):
-        self.logger.info(f"Installing MSEdgeDriver for channel {channel}")
+    def webdriver_supports_browser(self, webdriver_binary, browser_binary, browser_channel):
+        """Check that the browser binary and MSEdgeDriver versions are a valid match."""
+        browser_version = self.version(browser_binary)
+        edgedriver_version = self.webdriver_version(webdriver_binary)
 
-        if browser_binary is None:
-            browser_binary = self.find_binary(channel=channel)
-        else:
-            self.logger.info(f"Installing matching MSEdgeDriver for Edge binary at {browser_binary}")
+        if not edgedriver_version:
+            self.logger.warning("Unable to get version for MSEdgeDriver "
+                                f"{webdriver_binary}, rejecting it")
+            return False
 
-        version = self.version(browser_binary)
+        if not browser_version:
+            # If we can't get the browser version, we just have to assume the
+            # EdgeDriver is good.
+            return True
```

Comment:
I'm not certain if this is necessary (even for Chrome), but I remember leaving this around for Chrome because I was unsure if it was still needed. It might be worth at least adding a logger statement for when this logic is tripped (which should also be in the Chrome implementation 😅).

Reply:
Done! Thanks :)